libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -33,12 +33,32 @@ AP_BattMonitor_Backend::AP_BattMonitor_Backend(AP_BattMonitor &mon, AP_BattMonit
 /// capacity_remaining_pct - returns the % battery capacity remaining (0 ~ 100)
 uint8_t AP_BattMonitor_Backend::capacity_remaining_pct() const
 {
-    float mah_remaining = _params._pack_capacity - _state.consumed_mah;
-    if ( _params._pack_capacity > 10 ) { // a very very small battery
-        return MIN(MAX((100 * (mah_remaining) / _params._pack_capacity), 0), UINT8_MAX);
-    } else {
-        return 0;
+    float percent = 0;
+
+    if (has_current()) {
+        const float mah_remaining = _params._pack_capacity - _state.consumed_mah;
+        if ( _params._pack_capacity > 10 ) { // a very very small battery
+            percent = (100 * (mah_remaining) / _params._pack_capacity);
+        }
+
+    } else if (_params._voltage_full > 0 && _params._voltage_empty > 0) {
+        // well, we have no current but we at least know what the "full" and "empty" voltages are.
+        // Not nearly as good as a real current sensor but we can do the next best thing to create a percent
+        // NOTE: full can be less than empty which allows for an inverted analog sensor.
+
+        // TODO: take into account the non-linear voltage drop of a battery like the Beebop LUT
+        // https://www.richtek.com/battery-management/img/battery-discharge.png
+
+        percent = (100.0f * (_state.voltage - _params._voltage_empty)) / (_params._voltage_full - _params._voltage_empty);
```

Comment:
Possibly an edge case of operator error. But if full and empty are the same, you'll end up with a divide by zero. I actually have no idea what happens when you divide by zero in AP code. Also, would it be worth checking if we have voltage to begin with?

Reply:
good catch, that needs to be checked for. Thanks!